src/authenticode.c

```diff
@@ -233,14 +196,15 @@ static void parse_ms_countersig(PKCS7* p7, Authenticode* auth)
         int len = nested->value.sequence->length;
         const uint8_t* data = nested->value.sequence->data;
 
-        Countersignature* sig = ms_countersig_new(data, len, si->enc_digest);
-        if (!sig)
+        Countersignature* csig = ms_countersig_new(data, len, si->enc_digest);
+        if (!csig)
             return;
 
+        countersignature_array_insert(auth->countersigs, csig);
         /* Because MS TimeStamp countersignature has it's own SET of certificates
          * extract it back into parent signature for consistency with PKCS9 */
-        countersignature_array_insert(auth->countersigs, sig);
-        extract_ms_counter_certs(data, len, auth->certs);
+        if (csig->certs && csig->certs->count > 0)
```

Comment:
Just a small remark here, shouldn't the function called be rather responsible for these checks rather than caller? Just in case we need this somewhere else in the future, would this check have to be replicated in the new callsite?

Reply:
Honestly I am never 100% sure how to design some of these functions. Usually I feel like a caller should be responsible for not putting garbage into the function, but on the other hand, having that functionality within it makes more robust and cleaner code, for the expense of doing duplicate work sometimes. In our case cleaner code will be more of a priority.